fetools-app/src/pages/CharacterFinder.jsx

```diff
@@ -1,51 +1,104 @@
 import "../index.css";
 import React, { useState } from "react";
-import DisplayCategories from "../components/CharacterFinderComponents/DisplayCategories";
 import SearchField from "../components/SearchField";
-import DisplaySearchResults from "../components/CharacterFinderComponents/DisplaySearchResults";
 import ToolHeading from "../components/ToolsLayout/ToolHeading";
 import CharacterCategoryTab from "../components/CharacterFinderComponents/CharacterCategoryTab";
 import htmlCharacters from "../components/CharacterFinderComponents/htmlCharacters.json";
 import GoDeeper from "../components/ToolsLayout/GoDeeper";
+import CharacterCard from "../components/CharacterFinderComponents/CharacterCard";
 
 function CharacterFinder() {
-  const [selectedCategory, setSelectedCategory] = useState("symbols"); //placeholder for popular tab
+  const [selectedCategory, setSelectedCategory] = useState("popular");
   const [searchQuery, setSearchQuery] = useState("");
-  
+  const [searchResults, setSearchResults] = useState([]);
 
   const displayCharacters = (category) => {
     setSelectedCategory(category);
+    setSearchQuery("");
+    setSearchResults([]);
   };
 
+  const removeLetterVariations = (str) => {
+    return str.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
+  };
+
+  // ...
+
   const handleSearchChange = (query) => {
     setSearchQuery(query);
+
+    if (query.trim() === "") {
+      setSearchResults([]);
+      return;
+    }
+
+    const allCharacterArrays = [
+      ...htmlCharacters.letters,
+      ...htmlCharacters.punctuation,
+      ...htmlCharacters.numbers,
+      ...htmlCharacters.math,
+      ...htmlCharacters.currency,
+      ...htmlCharacters.arrows,
+      ...htmlCharacters.symbols,
+      ...htmlCharacters.emojis,
+    ];
+
+    let filteredResults;
+
+    if (query.length === 1) {
+      filteredResults = allCharacterArrays.filter(
+        (character) =>
+          character.character.toLowerCase() === query.toLowerCase() ||
+          removeLetterVariations(character.character.toLowerCase()) ===
+            removeLetterVariations(query.toLowerCase())
+      );
+
+      if (filteredResults.length === 0) {
+        filteredResults = allCharacterArrays.filter(
+          (character) =>
+            character.name.toLowerCase().includes(` ${query.toLowerCase()} `) ||
+            character.name
+              .toLowerCase()
+              .startsWith(`${query.toLowerCase()} `) ||
+            character.name.toLowerCase().endsWith(` ${query.toLowerCase()}`) ||
+            character.name.toLowerCase().includes(` ${query.toLowerCase()}`)
```

Comment:
```suggestion
            const name = character.name.toLowerCase();
            name.includes(` ${query.toLowerCase()}`) || name.startsWith(`${query.toLowerCase()} `);
```

These can be logically combined into two conditions.